firewall/handlers/whitelist_handler.go

```diff
@@ -8,124 +8,134 @@ import (
 			}
-
 		case *sqlparser.Update:
-
 		}
 	}
 
 
 	//Check rules
 	if len(handler.rules) != 0 {
-		violationOcurred, err := handler.testRulesViolation(query)
+		violationOccured, err := handler.testRulesViolation(query)
 		if err != nil {
-			return err
+			return ErrParseSqlRuleWhitelist
 		}
-		if violationOcurred {
-			return ErrForbiddenSqlStructure
+		if violationOccured {
+			return ErrForbiddenSqlStructureWhitelist
 		}
 	}
 	return nil
 }
 
-func(handler * WhitelistHandler) AddQueries(queries []string) {
+func (handler * WhitelistHandler) Reset(){
+
+	handler.queries = nil
+	handler.tables = nil
+	handler.rules = nil
+}
+
+func(handler * WhitelistHandler) AddQueries(queries []string) error {
 
 	for _, query := range queries {
-		handler.whiteQueries = append(handler.whiteQueries, query)
+		handler.queries = append(handler.queries, query)
+		_, err := sqlparser.Parse(query)
+		if err != nil {
+			return ErrQuerySyntaxError
+		}
 	}
-	handler.whiteQueries = removeDuplicates(handler.whiteQueries)
+	handler.queries = removeDuplicates(handler.queries)
+
+	return nil
 }
 
 func (handler * WhitelistHandler) RemoveQueries(queries []string) {
 
-	for _, query := range handler.whiteQueries {
-		yes, index := contains(handler.whiteQueries, query)
+	for _, query := range handler.queries {
+		yes, index := contains(handler.queries, query)
 		if yes {
-			handler.whiteQueries = append(handler.whiteQueries[:index], handler.whiteQueries[index+1:]...)
+			handler.queries = append(handler.queries[:index], handler.queries[index+1:]...)
 		}
 	}
 }
 
 func (handler * WhitelistHandler) AddTables(tableNames []string){
 	for _, tableName := range tableNames{
-		handler.allowedTables = append(handler.allowedTables, tableName)
+		handler.tables = append(handler.tables, tableName)
 	}
 
-	handler.allowedTables = removeDuplicates(handler.allowedTables)
+	handler.tables = removeDuplicates(handler.tables)
 }
 
 func (handler * WhitelistHandler) RemoveTables(tableNames []string){
 	for _, query := range tableNames{
-		yes, index := contains(handler.allowedTables, query)
+		yes, index := contains(handler.tables, query)
 		if yes {
-			handler.allowedTables = append(handler.allowedTables[:index], handler.allowedTables[index+1:]...)
+			handler.tables = append(handler.tables[:index], handler.tables[index+1:]...)
 		}
 	}
 }
 
-func (handler *WhitelistHandler) Refresh(){
-
-	handler.whiteQueries = nil
-	handler.allowedTables = nil
-	handler.rules = nil
-}
-
-func (handler *WhitelistHandler) AddRules(rules []string){
+func (handler * WhitelistHandler) AddRules(rules []string) error {
 	for _, rule := range rules{
 		handler.rules = append(handler.rules, rule)
+		_, err := sqlparser.Parse(rule)
+		if err != nil {
+			return ErrStructureSyntaxError
+		}
 	}
 
 	handler.rules = removeDuplicates(handler.rules)
+
+	return nil
 }
 
-func (handler *WhitelistHandler) RemoveRules(rules []string){
+func (handler * WhitelistHandler) RemoveRules(rules []string){
```

Comment:
`gofmt` will not agree with your space between `*` and type :)
please, run `go fmt ./...` in repo folder before pushing or we will not so excellent golang repo as now (https://goreportcard.com/report/github.com/cossacklabs/acra)  :((